src/include/storage/storage_structure/disk_overflow_file.h

```diff
@@ -37,19 +37,25 @@ class DiskOverflowFile : public StorageStructure {
         return copy;
     }
 
+    struct OverflowCache {
+        FileHandle* fileHandle = nullptr;
+        common::page_idx_t pageIdx = UINT32_MAX;
+        uint8_t* frame = nullptr;
+    };
+
     void readStringsToVector(
         transaction::TransactionType trxType, common::ValueVector& valueVector);
     void readStringToVector(transaction::TransactionType trxType, common::ku_string_t& kuStr,
-        common::InMemOverflowBuffer& inMemOverflowBuffer);
+        common::InMemOverflowBuffer& inMemOverflowBuffer, OverflowCache& overflowCache);
```

Comment:
that's a very good point! overloading `lookupString` so we can use it in `scanStringOverflow` without trying to cache (which is only additional overhead since it won't be triggered).